ehitchhiking-ui/src/app/pages/main-screen/yandex-map/yandex-map.component.ts

```diff
@@ -195,7 +210,7 @@ export class YandexMapComponent implements OnInit, OnChanges {
 			activeRoute.events.add('click', (event) => {
 				if (!this.myMap.balloon.isOpen()) {
```

Comment:
if (this.myMap.balloon.isOpen()) {
					return;
}

const coords = event.get('coords');
this.myMap.balloon.open(coords, YandexMapService.baloonInfo(data));
this.myMap.balloon.open(coords, YandexMapService.baloonInfo(this.routes[index]));

Reply:
pls move event handler to separate function

Reply:
left as is, explain the problem in more detail please